runtime/android/core_shell/src/org/xwalk/core/xwview/shell/XWalkViewShellActivity.java

```diff
@@ -359,4 +314,53 @@ public void onXWalkViewCreated(XWalkView view) {
         mProgressMap.put(view, 0);
         XWalkPreferences.setValue(XWalkPreferences.REMOTE_DEBUGGING, true);
     }
+
+    @Override
+    public void onXWalkReady() {
+        setContentView(R.layout.testshell_activity);
```

Comment:
It's okay to move the initialization of the activity to onCreate() to make the code easier to understand for someone don't know onXWalkReady() well. You can even create an XWalkView instance before onXWalkReady(), but just don't use it.